packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -255,6 +284,15 @@ export class CohereChatLanguageModel implements LanguageModelV2 {
 
             switch (type) {
               case 'content-start': {
+                if (value.delta.message.content.type === 'thinking') {
```

Comment:
use object

```suggestion
                const isThinking = value.delta.message.content.type === 'thinking'
                isReasoning[value.index] = isThinking
                if (isThinking) {
```



compare https://github.com/vercel/ai/blob/cdc6b7a612a2f775f5f6109fb80ef79f8e4f4989/packages/anthropic/src/anthropic-messages-language-model.ts#L894